sys/include/fido2/ctap/ctap_utils.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2021 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    fido2_ctap_utils FIDO2 CTAP utils
+ * @ingroup     fido2_ctap
+ * @brief       FIDO2 CTAP utility helper
+ *
+ * @{
+ *
+ * @file
+ * @brief       Definition for CTAP utility functions
+ *
+ * @author      Nils Ollrogge <nils.ollrogge@fu-berlin.de>
+ */
+
+#ifndef FIDO2_CTAP_CTAP_UTILS_H
+#define FIDO2_CTAP_CTAP_UTILS_H
+
+#include <stdint.h>
+#include "fido2/ctap/ctap.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief LED animation to indicate that user action is required
+ */
+void fido2_ctap_utils_led_animation(void);
+
+/**
+ * @brief Initialize button to be used for user presence test
+ *
+ * @return @ref ctap_status_codes_t
+ */
+int fido2_ctap_utils_init_gpio_pin(void);
+
+/**
+ * @brief Test user presence
+ *
+ * Successful if user clicks button in less than @ref CTAP_UP_TIMEOUT
+ *
+ * @return @ref ctap_status_codes_t
+ */
+int fido2_ctap_utils_user_presence_test(void);
+
+/**
+ * @brief Compare fido2 credentials based on creation time
+ *
+ * @param[in] k1    first resident key
+ * @param[in] k2    second resident key
+ *
+ * @return <0 if k2 has a bigger sign_count
+ * @return 0 if equal k1 and k2 have equal sign_count
+ * @return >0 if k1 has a bigger sign_count
+ */
+static inline int fido2_ctap_utils_cred_cmp(const void *k1, const void *k2)
```

Comment:
Where is this used?

Reply:
[Here](https://github.com/Ollrogge/RIOT/blob/6eb99790efb2042302b947dc91624bb623b9680c/sys/fido2/ctap/ctap.c#L1350)

Reply:
[Here](https://github.com/Ollrogge/RIOT/blob/6eb99790efb2042302b947dc91624bb623b9680c/sys/fido2/ctap/ctap.c#L1350)